packages/devtools_app/lib/src/inspector/inspector_tree_controller.dart

```diff
@@ -542,6 +585,151 @@ class InspectorTreeController extends Object {
       }
     }
   }
+
+  /* Search support */
+  @override
+  void previousMatch() {
+    super.previousMatch();
+    final currentSearchMatches = searchMatches.value;
+    for (int i = 0; i < currentSearchMatches.length; i++) {
+      final searchMatch = currentSearchMatches[i];
+
+      if (searchMatch.isSelected) {
+        final rowToSelect = i == 0
+            ? currentSearchMatches[currentSearchMatches.length - 1]
+            : currentSearchMatches[i - 1];
+
+        onSelectRow(rowToSelect);
+        break;
+      }
+    }
+  }
+
+  @override
+  void nextMatch() {
+    super.nextMatch();
+    final currentSearchMatches = searchMatches.value;
+    for (int i = 0; i < currentSearchMatches.length; i++) {
+      final searchMatch = currentSearchMatches[i];
+
+      if (searchMatch.isSelected) {
+        final rowToSelect = i == currentSearchMatches.length - 1
+            ? currentSearchMatches[0]
+            : currentSearchMatches[i + 1];
+
+        onSelectRow(rowToSelect);
+        break;
+      }
+    }
+  }
+
+  void dispose() {
+    if (_searchDebounce?.isActive ?? false) {
+      _searchDebounce.cancel();
+    }
+  }
+
+  Future _searchOperation;
+  Timer _searchDebounce;
+
+  @override
+  List<InspectorTreeRow> matchesForSearch(String search) {
+    updateMatches([]);
+    searchInProgress = true;
+
+    if (_searchDebounce?.isActive ?? false) {
+      _searchDebounce.cancel();
+    }
+    _searchDebounce =
+        Timer(Duration(milliseconds: search.isEmpty ? 0 : 300), () async {
+      // do something with query
+
+      // Abort any ongoing search operations and start a new one
+      if (_searchOperation != null) {
+        try {
+          await _searchOperation;
```

Comment:
I'm not seeing how this aborts the ongoing search operation? this looks like it actually waits for the ongoing operation to complete